### ISSUE TITLE
Plant Seedlings Classification using CNNs

### ISSUE THREAD

**Issue by keshripritesh** [PRIYANSHU2026/DataSentience-AIML#167]
<h3>📁 Project Name</h3>
<p><strong>Plant Seedlings Classification</strong></p>
<hr>
<h3>📖 Project Description</h3>
<p>This project aims to automate the classification of plant seedlings using various deep learning architectures. With the increase in demand for food crops and the limitations faced due to climate change, automating the classification of plant types at early stages can help in targeted interventions like selective weeding and optimized crop management.</p>
<p>This system leverages <strong>Convolutional Neural Networks (CNNs)</strong> to classify over 960 unique plant images belonging to 12 different species, using <strong>7 popular architectures</strong> like ResNet, AlexNet, VGG, Inception, MobileNet, DenseNet, and SqueezeNet.</p>
<hr>
<h3>🎯 Goals</h3>
<ul>
<li>
<p>Automate early-stage plant species classification.</p>
</li>
<li>
<p>Improve precision agriculture using AI.</p>
</li>
<li>
<p>Analyze and compare the performance of various CNN architectures.</p>
</li>
<li>
<p>Achieve high accuracy and reliability in classifying real-world crop data.</p>
</li>
</ul>
<hr>
<h3>📊 Dataset</h3>
<ul>
<li>
<p><strong>Name</strong>: V2 Plant Seedlings Dataset</p>
</li>
<li>
<p><strong>Source</strong>: <a href="https://www.kaggle.com/vbookshelf/v2-plant-seedlings-dataset">Kaggle - v2 Plant Seedlings</a></p>
</li>
<li>
<p><strong>Size</strong>: ~5,000 images, 12 species</p>
</li>
<li>
<p><strong>Format</strong>: RGB images categorized in folders by plant species</p>
</li>
</ul>
<hr>
<h3>🔧 Tech Stack</h3>
<ul>
<li>
<p><strong>Languages</strong>: Python</p>
</li>
<li>
<p><strong>Frameworks</strong>: TensorFlow, PyTorch, Keras</p>
</li>
<li>
<p><strong>Libraries</strong>: NumPy, Pandas, Matplotlib, Seaborn, Time, OS, Shutil</p>
</li>
<li>
<p><strong>Jupyter Notebook</strong> for code documentation and visualization</p>
</li>
</ul>
<hr>
<h3>🚀 Models Used</h3>

Model | Accuracy
-- | --
ResNet | 69%
AlexNet | 69%
Inception | 68%
VGG | 68%
MobileNet | 67%
DenseNet | 67%
SqueezeNet | 67%

<hr>
<h3>📈 Visuals</h3>
<ul>
<li>
<p>Confusion Matrix</p>
</li>
<li>
<p>Training/Validation Accuracy &amp; Loss Graphs</p>
</li>
<li>
<p>Architecture-wise Accuracy Comparison Table</p>
</li>
</ul>
<hr>
<h3>✅ Tasks Checklist</h3>
<ul class="contains-task-list">
<li class="task-list-item">
<p><input type="checkbox" checked="checked" disabled="disabled"> Dataset preprocessing and loading</p>
</li>
<li class="task-list-item">
<p><input type="checkbox" checked="checked" disabled="disabled"> Model training and evaluation</p>
</li>
<li class="task-list-item">
<p><input type="checkbox" checked="checked" disabled="disabled"> Performance visualization</p>
</li>
<li class="task-list-item">
<p><input type="checkbox" checked="checked" disabled="disabled"> Architecture comparison</p>
</li>
<li class="task-list-item">
<p><input type="checkbox" checked="checked" disabled="disabled"> Final accuracy reporting</p>
</li>
</ul>
<hr>
<h3>🧑‍💻 Author</h3>
<p><strong>Abhishek Sharma</strong><br>
GitHub: <a href="https://github.com/abhisheks008">@abhisheks008</a><br>
Event: #LGMSOC21</p>
<hr>
<h3>📝 License</h3>
<p><a>MIT License</a></p>
<hr>
<h3>🙌 Additional Notes</h3>
<ul>
<li>
<p>This project is a good fit for contributors who are enthusiastic about deep learning, plant science, and computer vision.</p>
</li>
<li>
<p>Ideal for beginners to intermediate learners looking to contribute to impactful real-world applications.</p>
</li>
</ul>
<hr>
<p>Let me know if you'd like this adapted into a <code>README.md</code> or Pull Request message too!</p></body></html>
</body>
</html>Here is a detailed GitHub issue template you can use to **add your Plant Seedlings Classification project** to the **Social Summer of Code (SSoC)**:

---

**Title: 🚀 Add New Project: Plant Seedlings Classification using CNNs**

---

### 📌 Project Type

* [x] Machine Learning
* [x] Deep Learning
* [x] Image Classification
* [x] Agriculture Tech

---

### 📁 Project Name

**Plant Seedlings Classification**

---

### 📖 Project Description

This project aims to automate the classification of plant seedlings using various deep learning architectures. With the increase in demand for food crops and the limitations faced due to climate change, automating the classification of plant types at early stages can help in targeted interventions like selective weeding and optimized crop management.

This system leverages **Convolutional Neural Networks (CNNs)** to classify over 960 unique plant images belonging to 12 different species, using **7 popular architectures** like ResNet, AlexNet, VGG, Inception, MobileNet, DenseNet, and SqueezeNet.

---

### 🎯 Goals

* Automate early-stage plant species classification.
* Improve precision agriculture using AI.
* Analyze and compare the performance of various CNN architectures.
* Achieve high accuracy and reliability in classifying real-world crop data.

---

### 📊 Dataset

* **Name**: V2 Plant Seedlings Dataset
* **Source**: [[Kaggle - v2 Plant Seedlings](https://www.kaggle.com/vbookshelf/v2-plant-seedlings-dataset)](https://www.kaggle.com/vbookshelf/v2-plant-seedlings-dataset)
* **Size**: \~5,000 images, 12 species
* **Format**: RGB images categorized in folders by plant species

---

### 🔧 Tech Stack

* **Languages**: Python
* **Frameworks**: TensorFlow, PyTorch, Keras
* **Libraries**: NumPy, Pandas, Matplotlib, Seaborn, Time, OS, Shutil
* **Jupyter Notebook** for code documentation and visualization

---

### 🚀 Models Used

| Model      | Accuracy |
| ---------- | -------- |
| ResNet     | 69%      |
| AlexNet    | 69%      |
| Inception  | 68%      |
| VGG        | 68%      |
| MobileNet  | 67%      |
| DenseNet   | 67%      |
| SqueezeNet | 67%      |

---

### 📈 Visuals

* Confusion Matrix
* Training/Validation Accuracy & Loss Graphs
* Architecture-wise Accuracy Comparison Table

---

### ✅ Tasks Checklist

* [x] Dataset preprocessing and loading
* [x] Model training and evaluation
* [x] Performance visualization
* [x] Architecture comparison
* [x] Final accuracy reporting

---

### 🙌 Additional Notes

* This project is a good fit for contributors who are enthusiastic about deep learning, plant science, and computer vision.
* Ideal for beginners to intermediate learners looking to contribute to impactful real-world applications.

---

**Comment by PRIYANSHU2026** [comment]
ok @keshripritesh

**Comment by Divodude** [comment]
Hi! I’d like to work on this. Is it still available?

**Comment by student-smritipandey** [comment]
@PRIYANSHU2026  i want to contribute. is it available?
